python/flip.py

```diff
@@ -31,584 +30,10 @@
-
-			# Compute HDR-FLIP and exposure map
-			t0 = time.time()
-			flip, exposure_map = compute_hdrflip(reference,
-												 test,
-												 directory=args.directory,
-												 reference_filename=reference_filename,
-												 test_filename=test_filename,
-												 basename=basename,
-												 default_basename=default_basename,
-												 pixels_per_degree=pixels_per_degree,
-												 tone_mapper=tone_mapper,
-												 start_exposure=start_exposure,
-												 stop_exposure=stop_exposure,
-												 num_exposures=num_exposures,
-												 save_ldr_images=args.save_ldr_images,
-												 save_ldrflip=args.save_ldrflip,
-												 no_magma=args.no_magma)
-			t = time.time()
-
-			# Store results
-			flip_array[:, :, idx] = flip
-
-			# Save exposure map
-			if args.no_exposure_map == False:
-				exposure_map_path = ("%s/%s.png") % (args.directory, "exposure_map." + basename if default_basename else basename + ".exposure_map")
-				save_image(exposure_map_path, exposure_map)
-
-		else:
-			test = load_image_array(test_path)
-			assert reference.shape == test.shape
-			if (reference < 0).any() or (reference > 1).any() or (test < 0).any() or (test > 1).any():
-				reference = np.clip(reference, 0.0, 1.0)
-				test = np.clip(test, 0.0, 1.0)
-				if args.verbosity == 3:
-					print('=============================================================================================')
-					print('WARNING: either reference or test (or both) images have pixel component values outside [0,1].')
-					print('LDR-FLIP is defined only for [0, 1]. Values have been clamped.')
-					print('=============================================================================================')
-			reference, test = check_nans(reference, test, args.verbosity)
-
-			# Compute LDR-FLIP
-			t0 = time.time()
-			flip = compute_ldrflip(reference,
-								   test,
-								   pixels_per_degree=pixels_per_degree
-								   ).squeeze(0)
-			t = time.time()
-			flip_array[:, :, idx] = flip
-
-		if args.verbosity > 0:
-			print("FLIP between reference image <%s.%s> and test image <%s.%s>:" % (reference_filename, image_format, test_filename, image_format))
-
-		# Output pooled values
-		evaluation_time = t - t0
-		print_pooled_values(flip, hdr, args.textfile, args.csv, args.directory, basename, default_basename, args.reference, test_path, evaluation_time, args.verbosity)
-
-		# Print time spent computing FLIP
-		if args.verbosity > 1: print(("\tEvaluation time: %.4f seconds") % evaluation_time)
-		if (args.verbosity > 0 and idx < (num_test_images - 1)): print("")
-
-		# Save FLIP map
-		error_map_path = ("%s/%s%s.png") % (args.directory, "flip." if default_basename else "", basename)
-		if args.no_error_map == False:
-			if args.no_magma == True:
-				error_map = flip
-			else:
-				error_map = CHWtoHWC(index2color(np.round(flip * 255.0), get_magma_map()))
-			save_image(error_map_path, error_map)
+    flip.run_tool(sys.argv)
```

Comment:
If you change to `FLIPToolHelpers::execute` (instead of `main`), then perhaps use `flip.execute()` here as well?

Reply:
Fixed!